translation/transport.go

```diff
@@ -65,24 +69,71 @@ func ConfigHandler(c *Options) {
 		Methods(http.MethodDelete, http.MethodPut, http.MethodPost)
 }
 
-/* Request Decoding */
+const (
+	PartnerIdHeader               = "X-Xmidt-Partner-Id"
+)
+// getPartnerIDs returns the array that represents the partner-ids that were
+// passed in as headers.  This function handles multiple duplicate headers.
+func getPartnerIDs(h http.Header) []string {
+	headers, ok := h[PartnerIdHeader]
+	if !ok {
+		return nil
+	}
+
+	var partners []string
+
+	for _, value := range headers {
+		fields := strings.Split(value, ",")
+		for i := 0; i < len(fields); i++ {
+			fields[i] = strings.TrimSpace(fields[i])
+		}
+		partners = append(partners, fields...)
+	}
 
+	return partners
+}
+
+
+/* Request Decoding */
 func decodeRequest(ctx context.Context, r *http.Request) (decodedRequest interface{}, err error) {
 	var (
 		payload []byte
 		wrpMsg  *wrp.Message
 	)
-
 	if payload, err = requestPayload(r); err == nil {
 		var tid = ctx.Value(common.ContextKeyRequestTID).(string)
-		if wrpMsg, err = wrap(payload, tid, mux.Vars(r)); err == nil {
-			decodedRequest = &wrpRequest{
-				WRPMessage:      wrpMsg,
-				AuthHeaderValue: r.Header.Get(authHeaderKey),
+		auth, ok := bascule.FromContext(ctx)
```

Comment:
```suggestion
		var partnerIDs []string
		auth, ok := bascule.FromContext(ctx)
```

I think the code will be more organized if you set this partnerIDs variable throughout the if statements and then call `wrap()` one time after the logic to the determine the partnerIDs.